src/core/thread/mesh_forwarder.cpp

```diff
@@ -1102,9 +1102,12 @@ void MeshForwarder::HandleFragment(const uint8_t *         aFrame,
         message->SetNetworkTimeOffset(aLinkInfo.mNetworkTimeOffset);
 #endif
 
-        // Security Check
         VerifyOrExit(Get<Ip6::Filter>().Accept(*message), error = OT_ERROR_DROP);
 
+#if OPENTHREAD_FTD
+        CheckReachability(*message, aMacSource, aMacDest);
```

Comment:
Do I get it correctly? We don't care about returned `error` here (we want to send the unreachable error only). 
(can we add `IgnoreReturnValue` here).

Reply:
Changed to return void. Also renamed to `SendIcmpErrorIfDstUnreach()`.